features/topology-map/plugins/org.opennms.features.topology.plugins.topo.bsm/src/main/java/org/opennms/features/topology/plugins/topo/bsm/ReductionKeyVertex.java

```diff
@@ -44,12 +49,23 @@ public ReductionKeyVertex(GraphVertex graphVertex) {
     }
 
     protected ReductionKeyVertex(String reductionKey, int level, Status status) {
-        super(Type.ReductionKey + ":" + reductionKey, reductionKey, level, status);
+        super(Type.ReductionKey + ":" + reductionKey, getLabelFromReductionKey(reductionKey), level, status);
         this.reductionKey = reductionKey;
         setTooltipText(String.format("Reduction Key '%s'", reductionKey));
         setIconKey("bsm.reduction-key");
     }
 
+    protected static String getLabelFromReductionKey(String reductionKey) {
+        String label;
+        Matcher m = REDUCTION_KEY_LABEL_PATTERN.matcher(reductionKey);
+        if (m.matches()) {
+            label = String.format("%s:%s", m.group(1), m.group(2));
+        } else {
+            label = reductionKey;
+        }
+        return label.substring(0, Math.min(label.length(), MAX_LABEL_LENGTH));
```

Comment:
We can limit the size of the text box with CSS, which isn't ideal. I'm not aware of any way of doing a substring per say.